code/game/machinery/fusion_engine.dm

```diff
@@ -145,97 +145,102 @@
 			to_chat(user, "<span class='warning'>You need to remove the fuel cell from [src] first.</span>")
 			return TRUE
 		return TRUE
-	else if(iswelder(O))
-		if(buildstate == FUSION_ENGINE_HEAVY_DAMAGE)
-			var/obj/item/tool/weldingtool/WT = O
-			if(WT.remove_fuel(1, user))
-				if(user.mind && user.mind.cm_skills && user.mind.cm_skills.engineer < SKILL_ENGINEER_ENGI)
-					user.visible_message("<span class='notice'>[user] fumbles around figuring out [src]'s internals.</span>",
-					"<span class='notice'>You fumble around figuring out [src]'s internals.</span>")
-					var/fumbling_time = 100 - 20 * user.mind.cm_skills.engineer
-					if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)))
-						return FALSE
-				playsound(loc, 'sound/items/weldingtool_weld.ogg', 25)
-				user.visible_message("<span class='notice'>[user] starts welding [src]'s internal damage.</span>",
-				"<span class='notice'>You start welding [src]'s internal damage.</span>")
-				if(do_after(user, 200, TRUE, src, BUSY_ICON_BUILD, extra_checks = CALLBACK(WT, /obj/item/tool/weldingtool/proc/isOn)))
-					if(buildstate != FUSION_ENGINE_HEAVY_DAMAGE || is_on)
-						return FALSE
-					playsound(loc, 'sound/items/Welder2.ogg', 25, 1)
-					buildstate = FUSION_ENGINE_MEDIUM_DAMAGE
-					user.visible_message("<span class='notice'>[user] welds [src]'s internal damage.</span>",
-					"<span class='notice'>You weld [src]'s internal damage.</span>")
-					update_icon()
-					return TRUE
-			else
-				to_chat(user, "<span class='warning'>You need more welding fuel to complete this task.</span>")
-				return FALSE
-	else if(iswirecutter(O))
-		if(buildstate == FUSION_ENGINE_MEDIUM_DAMAGE && !is_on)
+	else
+		return ..()
+
+/obj/machinery/power/fusion_engine/welder_act(mob/living/user, obj/item/O)
```

Comment:
Same here

Reply:
It was meant to be here. The first proper one is going to be larger in scale.